ebpf/state.go

```diff
@@ -287,119 +297,49 @@ func (ns *networkState) RemoveClient(clientID string) {
 	delete(ns.clients, clientID)
 }
 
-func (ns *networkState) removeExpiredClients(now time.Time, removeExpiredStats bool) {
+func (ns *networkState) cleanupState(now time.Time, clearExpiredStats, flushStats bool) {
 	ns.Lock()
 	defer ns.Unlock()
 
+	// Remove expired clients + stats
 	deletedStats := 0
 	for id, c := range ns.clients {
 		if c.lastFetch.Add(ns.expiry).Before(now) {
 			delete(ns.clients, id)
-		} else if removeExpiredStats { // Look for inactive stats objects and remove them
+		} else if clearExpiredStats { // Look for inactive stats objects and remove them
 			deletedStats += ns.removeExpiredStats(c, now)
 		}
 	}
 
 	if deletedStats > 0 {
 		log.Debugf("removed %d expired stats objects in %d", deletedStats, time.Now().Sub(now))
 	}
+
+	if flushStats {
```

Comment:
Is there any reason this bit needs to be part of `cleanupState` ? The two functions are unrelated, except that they both get called periodically.

Reply:
Technically we're clearing the telemetry map here as well. I'd prefer to leave it here instead of having another independent goroutine doing this work.